### DISCUSSION TITLE
Should SL4FJ `addKeyValue` method usage work?

### DISCUSSION THREAD

**Discussion by dtanner** [logfellow/logstash-logback-encoder#945] [Q&A] [answered]
Hi,

I'm trying to only use SLF4J interface methods in my code, yet still be able to produce JSON output with custom key/value fields.

Using the logstash logback library-specific `StructuredArguments` class works as expected (`"somekey"` field is output):
```kotlin
logger.atInfo().log("kvmessage", StructuredArguments.keyValue("somekey", "somevalue"))
```

But using the [sl4fj method](https://www.slf4j.org/apidocs/org/slf4j/spi/LoggingEventBuilder.html#addKeyValue-java.lang.String-java.lang.Object-) doesn't appear to work with the logback-logstash encoder. (no `"somekey"` field is output):
```kotlin
logger.atInfo().addKeyValue("somekey", "somevalue").log("kvmessage")
```

Is there a technique I can use in the code that does work?  FWIW I've also considered Markers, but the problem there is that SLF4J doesn't provide a usable Marker Map implementation out of the box.  Thanks for any ideas.

**Comment by philsttr** [comment, marked as answer]
Hey @dtanner,

Support for slf4j 2's fluent api key value pairs was added in [logstash-logback-encoder 7.4](https://github.com/logfellow/logstash-logback-encoder/releases/tag/logstash-logback-encoder-7.4)
